Automattic/wp-calypso
# Site Level User Profile: use site-level language (and admin color) on Calypso pages

> [!CAUTION]
> DO NOT MERGE this PR yet until the Jetpack [PR](https://github.com/Automattic/jetpack/pull/38879) is deployed to Atomic. (But you can still review.)

Related to:

- https://github.com/Automattic/dotcom-forge/issues/7754

## Proposed Changes

This PR makes site-specific Calypso pages to use the site-level user language which set in `wp-admin/profile.php`.

The logic is very similar to:

- https://github.com/Automattic/wp-calypso/pull/89261

This PR makes use of the newly created `/wpcom/v2/profile` endpoint, which returns the user's locale (along with the admin color). Then, it sets the current locale based on the response.

Later, we will clean up the now-unused `/wpcom/v2/admin-color` endpoint as it's merged to the above new endpoint.

|Type|Before|After|
|-|-|-|
|Atomic<br>Classic|<img width="843" alt="image" src="https://github.com/user-attachments/assets/c42c5509-4335-440d-ad2e-abafa410a76d">|<img width="843" alt="image" src="https://github.com/user-attachments/assets/4dc6f185-061b-4760-82ff-019788252173">|
|Simple<br>Classic|<img width="843" alt="image" src="https://github.com/user-attachments/assets/36fc7e03-cf2f-4299-8475-d5ea2c6c9c50">|<img width="843" alt="image" src="https://github.com/user-attachments/assets/0a14f479-3f96-44d1-8857-3a370fd8b050">|
|Atomic<br>Default|<img width="850" alt="image" src="https://github.com/user-attachments/assets/725e8931-bfb8-4bb2-bca3-9cc92f4c4ecd">|<img width="848" alt="image" src="https://github.com/user-attachments/assets/8a80b2c6-1974-4b01-8e31-a5d9495e39b3">|
|Simple<br>Default|<img width="843" alt="image" src="https://github.com/user-attachments/assets/ffe37ad2-0cb3-4eb4-aa2d-167f57a07c9e">|<img width="843" alt="image" src="https://github.com/user-attachments/assets/d2e99455-87ce-4b86-9506-57cffd1786af">

(Atomic Default is pending https://github.com/Automattic/dotcom-forge/issues/8511).
## Why are these changes being made?


We want to expose `profile.php` settings to all types of sites.

## Testing Instructions

> [!WARNING]
> !!! IMPORTANT: USE THE CALYPSO LIVE URL FROM THIS PR INSTEAD !!!:
> - https://github.com/Automattic/wp-calypso/pull/93733


### Test on a Simple Classic site

1. Prepare a Simple Classic site.
8. Go to your site's `/wp-admin/profile.php`:
    - Change the admin color to a color other than the one you set in `/me/account`.
    - Change the user language to a language other than the one you set in `/me/account`.
10. Go to `<calypso live URL>/home/<site slug>`:
    - VERIFY that the color is the same as the color you set previously.
    - VERIFY that the language is the same as the language you set in step 8.
1. Don't do anything on your sandbox, and perform the next sections below.

### Test on a Simple Default site

1. Prepare a Simple Default site.
1. Go to `<calypso live URL>/home/<site slug>`:
    - VERIFY that it's still using global admin color and langauge.
1. Now, on your sandbox, run:
    - `wpsh`
    - `update_blog_option( <your blog id>, 'wpcom_site_level_user_profile', '1' );`
1. Follow the same instructions as above from step (4) onwards.

### Test on an Atomic Classic site

1. Prepare an Atomic Classic site.
1. Use Bleeding Edge for the Jetpack plugin via Jetpack Beta Tester, to pick up this PR https://github.com/Automattic/jetpack/pull/38879
1. Follow the same instructions as above from step (4) onwards.

### Test on an Atomic Default site

1. Prepare an Atomic Default site.
1. Use Bleeding Edge for the Jetpack plugin via Jetpack Beta Tester, to pick up this PR https://github.com/Automattic/jetpack/pull/38879
2. Go to `_cli` then run `wp option update wpcom_site_level_user_profile 1`
1. Follow the same instructions as above from step (4) onwards.


## Pre-merge Checklist
